rbac/internal/views.py

```diff
@@ -898,3 +898,56 @@ def username_lower(request):
             )
         Principal.objects.bulk_update(principals, ["username"])
         return HttpResponse(f"Updated {len(principals)} usernames", status=200)
+
+
+def principal_removal(request):
+    """Get/Delete not active principals.
+
+    GET or DELETE /_private/api/utils/principal/?usernames=a,b,c
+    """
+    logger.info(f"Principal edit or removal: {request.method} {request.user.username}")
+    if request.method not in ["DELETE", "GET"]:
+        return HttpResponse('Invalid method, only "DELETE" or "GET" is allowed.', status=405)
+
+    if not request.GET.get("usernames"):
+        return HttpResponse("Please provided a list of usernames with comma separated.", status=400)
+    usernames = request.GET.get("usernames").split(",")
+    resp = PROXY.request_filtered_principals(usernames, org_id=None, options={"return_id": True})
+
+    if isinstance(resp, dict) and "errors" in resp:
+        return HttpResponse(resp.get("errors"), status=400)
+
+    active_users = {(principal_data["username"], principal_data["org_id"]) for principal_data in resp["data"]}
+
+    principals = (
+        Principal.objects.filter(username__in=usernames).filter(type=Principal.Types.USER).prefetch_related("tenant")
+    )
+    principals_delete = [
+        principal for principal in principals if (principal.username, principal.tenant.org_id) not in active_users
+    ]
+
+    principal_usernames = [principal.username for principal in principals_delete]
+
+    if request.method == "GET":
+        return HttpResponse(
+            f"Principals to be deleted: {principal_usernames}",
+            status=200,
+        )
+    if not destructive_ok("api"):
+        return HttpResponse("Destructive operations disallowed.", status=400)
+
+    with transaction.atomic():
+        bootstrap_service = V2TenantBootstrapService(OutboxReplicator())
+        for principal in principals_delete:
+            if not principal.user_id:
+                principal.delete()
+            else:
+                user = User()
```

Comment:
@astrozzc what is reason for this branch ? (Maybe I am missing some context.)

Reply:
Should we keep those users if they don't exist in IT service ?

Reply:
There are some users not active anymore, we should remove them. They are attached to some custom group, would cause problem when we run migrator because they don't have user_id.